src/NuGet.Core/NuGet.Common/PathUtil/DirectoryUtility.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace NuGet.Common
+{
+    /// <summary>
+    /// Directory operation helpers.
+    /// </summary>
+    public static class DirectoryUtility
+    {
+        private static object s_sharedFolderCreationLock = new object();
+
+        /// <summary>
+        /// Creates all directories and subdirectories in the specified path unless they already exist.
+        /// New directories can be read and written by all users.
+        /// </summary>
+        public static void CreateSharedDirectory(string path)
+        {
+            if (RuntimeEnvironmentHelper.IsWindows)
```

Comment:
It is common for Linux users to work on the same machine, probably that is why reports are only coming from them. I'd assume the fix is also valid for Mac.
I don't think Mono has something to do with this, unless IsWindows=false when mono runs on Windows.